### PR TITLE
add field/2 for extracting field from struct

### PR DESCRIPTION
**Pull request by lkarthee**
```elixir
iex> s = Series.from_list([%{a: 1}, %{a: 2}])
iex> Series.field(s, "a")
#Explorer.Series<
  Polars[2]
  s64 [1, 2]
>
iex> Series.field(s, "b")
* (ArgumentError) field "b" not found in fields ["a"]
    (explorer 0.9.0-dev) lib/explorer/series.ex:6041: Explorer.Series.field/2
    iex:3: (file)

iex> df = DF.new([%{a: %{n: 1}}, %{a: %{n: 1}}])
iex> DF.mutate(df, n: field(a, "n"))
#Explorer.DataFrame<
  Polars[2 x 2]
  a struct[1] [%{"n" => 1}, %{"n" => 1}]
  n s64 [1, 1]
>
iex> DF.mutate(df, n: field(a, "m"))
** (ArgumentError) field "m" not found in fields ["n"]
    (explorer 0.9.0-dev) lib/explorer/series.ex:6041: Explorer.Series.field/2
    iex:3: (file)
    iex:3: (file)
```

### CODE REVIEW

**Review by josevalim — APPROVED** [review]

File: lib/explorer/series.ex [hunk 1]
```diff
@@ -6020,6 +6020,29 @@ defmodule Explorer.Series do
...
+      iex> s = Series.from_list([%{a: 1}, %{a: 2}])
+      iex> Series.field(s, "a")
+      #Explorer.Series<
+        Polars[2]
+        s64 [1, 2]
+      >
+  """
+  @doc type: :struct_wise
+  @spec field(Series.t(), Explorer.Backend.Series.valid_types()) :: Series.t()
+  def field(%Series{dtype: {:struct, dtype}} = series, name) do
```
Comment (josevalim) [inline lib/explorer/series.ex:6037]: Aren't struct fields all strings?

```suggestion
  @spec field(Series.t(), String.t()) :: Series.t()
  def field(%Series{dtype: {:struct, dtype}} = series, name) when is_binary(name) do
```

**Comment by lkarthee** [comment]
Mistakenly clicked some button on mobile app. Will send pr again.

**Comment by josevalim** [comment]
I can reopen. Feel free to close this one if you prefer it closed. :D

**Comment by lkarthee** [comment]
I prefer a merge :D 

I need few struct features for migrating to pure explorer workflow. I have a json column in database and I am using polars to extract the data. 
- field
- json_decode
- json_path_match
- json_encode ?? 

One PR at a time..

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**